pkg/p2p/libp2p/connections_test.go

```diff
@@ -201,7 +203,15 @@ func TestStreamsMaxIncomingLimit(t *testing.T) {
 	expectPeers(t, s2, overlay1)
 	expectPeersEventually(t, s1, overlay2)
 
-	maxIncomingStreams := goyamux.DefaultTransport.MaxIncomingStreams
+	var limits rcmgr.Limit
+	if err := s2.Host().Network().ResourceManager().ViewTransient(func(scope network.ResourceScope) error {
+		limits = scope.(rcmgr.ResourceScopeLimiter).Limit()
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	maxIncomingStreams := uint32(limits.GetStreamLimit(network.DirInbound))
```

Comment:
It looks like that the default limit is 128, but currently `goyamux.DefaultTransport.MaxIncomingStreams` is set to 5000. Maybe with new libp2p version it is ok to have a lower number, but has to be validated.

Reply:
Yes, I agree here. Let's start with this baseline and we can tweak the parameters later.

Reply:
I think it would be good to have a metric inside `libp2p` package on the protocol level that intercepts the error returned from a protocol handler to see whether it is `network.ErrReset` and if so increment a counter. This would allow us to know whether a certain change creates elevated stream reset conditions.

Reply:
Done.